EIPS/eip-6806.md

```diff
@@ -0,0 +1,130 @@
+---
+eip: 6806
+title: ERC-721 Holding Time Tracking
+description: Add holding time information to ERC-721 tokens
+author: Saitama (@saitama2009), Combo <combo@1combo.io>, Luigi <luigi@1combo.io>
+discussions-to: https://ethereum-magicians.org/t/draft-eip-erc721-holding-time-tracking/13605
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-03-30
+requires: 721
+---
+
+## Abstract
+
+This standard is an extension of [ERC-721](./eip-721.md). It adds an interface that tracks and describes the holding time of a Non-Fungible Token (NFT) by an account. 
+
+## Motivation
+
+In some use cases, it is valuable to know the duration for which a NFT has been held by an account. This information can be useful for rewarding long-term holders, determining access to exclusive content, or even implementing specific business logic based on holding time. However, the current ERC-721 standard does not have a built-in mechanism to track NFT holding time.
+
+This proposal aims to address these limitations by extending the ERC-721 standard to include holding time tracking functionality.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+**Interface**
+
+The following interface extends the existing ERC-721 standard:
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.0
+
+interface IERC6806 {
+    function getHoldingInfo(
+        uint256 tokenId
+    ) external view returns (address holder, uint256 holdingTime);
+}
+```
+
+**Functions**
+
+### getHoldingInfo
+
+```
+function getHoldingInfo(uint256 tokenId) external view returns (address holder, uint256 holdingTime);
+```
+
+This function returns the current holder of the specified NFT and the length of time (in seconds) the NFT has been held by the current account.
+
+* `tokenId`: The unique identifier of the NFT.
+* Returns: A tuple containing the current holder's address and the holding time (in seconds).
+
+## Rationale
+
+The addition of the `getHoldingInfo` function to an extension of the ERC-721 standard enables developers to implement NFT-based applications that require holding time information. This extension maintains compatibility with existing ERC-721 implementations while offering additional functionality for new use cases.
+
+The `getHoldingInfo` function provides a straightforward method for retrieving the holding time and holder address of an NFT. By using seconds as the unit of time for holding duration, it ensures precision and compatibility with other time-based functions in smart contracts.
+
```

Comment:
How about something like this, to explain the return values of `getHoldingInfo`?

```suggestion

`getHoldingInfo` returns both `holder` and `holdingTime` so that some token owners (as decided by the implementation) can be ignored for the purposes of calculating holding time. For example, a contract may take ownership of an NFT as collateral for a loan. Such a loan contract could be ignored, so the real owner's holding time increases properly.
```

Reply:
Ok, it looks good